### DISCUSSION TITLE
How to add new camera to the scene?

### DISCUSSION THREAD

**Discussion by kefhuang** [empriselab/RCareWorld#168] [Other Q&A] [answered]
Hello, I noticed that in the code comments, it mentions that an unlimited number of cameras can be used. However, I wasn't able to find any functions or APIs to add additional cameras beyond the default one.

Could you provide some guidance on how to add extra cameras to the environment?

**Comment by YoruCathy** [comment]
Check this https://github.com/empriselab/RCareWorld/blob/72af472b9da2b0afff80ca3440c3db9acf494cd1/pyrcareworld/pyrcareworld/demo/examples/example_camera.py#L35

**Reply by kefhuang** [reply]
Hi, Thanks we have tried it out before. Here is our code
```
env = BathingEnv(graphics=True, assets=["Camera"])
new_cam = env.InstanceObject(name="Camera", id=123456, attr_type=attr.CameraAttr)
new_cam.SetTransform(position=[0, 1.7, 0], rotation=[90, 0, 0])

new_cam.GetRGB(512, 256)
env.step()
rgb = np.frombuffer(new_cam.data["rgb"], dtype=np.uint8)
```
The `new_cam.data` has nothing it. Is there anything wrong with our implementation?

**Reply by YoruCathy** [reply]
Try this 
```
env = BathingEnv(graphics=True, assets=["Camera"])
new_cam = env.InstanceObject(name="Camera", id=123456, attr_type=attr.CameraAttr)
new_cam.SetTransform(position=[0, 1.7, 0], rotation=[90, 0, 0])

new_cam.GetRGB(512, 512)
env.step()
new_cam = np.frombuffer(camera.data["rgb"], dtype=np.uint8)
new_cam = cv2.imdecode(rgb, cv2.IMREAD_COLOR)
new_cam.imwrite("rgb.png", rgb)
```

**Reply by kefhuang** [reply]
not sure if this is you typo `new_cam = np.frombuffer(camera.data["rgb"], dtype=np.uint8)` but i changed it to `rgb = np.frombuffer(new_cam.data["rgb"], dtype=np.uint8)` and got the error
```
  File "/workspaces/ralla_care/scripts/test_bathing.py", line 148, in <module>
    rgb = np.frombuffer(new_cam.data["rgb"], dtype=np.uint8)
KeyError: 'rgb'
```

Also we looked into the `new_cam.data` it is empty. 

Here is the code i run for your reference
```
env = BathingEnv(graphics=True, assets=["Camera"])
new_cam = env.InstanceObject(name="Camera", id=123456, attr_type=attr.CameraAttr)
new_cam.SetTransform(position=[0, 1.7, 0], rotation=[90, 0, 0])

new_cam.GetRGB(512, 512)
env.step()
rgb = np.frombuffer(new_cam.data["rgb"], dtype=np.uint8)
rgb = cv2.imdecode(rgb, cv2.IMREAD_COLOR)
cv2.imwrite("rgb.png", rgb)
```

**Reply by YoruCathy** [reply, marked as answer]
The code seems fine. We'll look into this on our end. 
Actually, have you tried making a copy of the current camera? This is the API: https://github.com/empriselab/RCareWorld/blob/76f0bb9bf6c76d39ca985d8bd06769c921b57a46/pyrcareworld/pyrcareworld/attributes/base_attr.py#L197

**Reply by kefhuang** [reply]
Oh Copy works. 
Thanks. Appreciate it.

**Reply by thisjustin123** [reply]
Found out why this is happening, will push a fix soon! After the fix, your original code should work. @kefhuang 

#170
